crates/ruff/src/rules/flake8_pyi/rules/simple_defaults.rs

```diff
@@ -97,6 +97,36 @@ impl Violation for UnassignedSpecialVariableInStub {
     }
 }
 
+/// ## What it does
+/// Checks for `typehint.TypeAlias` annotation in type aliases.
+///
+/// ## Why is this bad?
+/// It makes harder for someone reading the code to differentiate
+/// between a value assignment and a type alias.
+///
+/// ## Example
+/// ```python
+/// IntOrStr = int | str
+/// ```
+///
+/// Use instead:
+/// ```python
+/// IntOrStr: TypeAlias = int | str
+/// ```
+#[violation]
+pub struct TypeAliasCheck {
+    name: String,
+    value: String,
+}
+
+impl Violation for TypeAliasCheck {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let TypeAliasCheck { name, value } = self;
+        format!("Use `typing.TypeAlias` for type aliases in `{name}`, e.g. \"{name}: typing.TypeAlias = {value}\"")
```

Comment:
I'd either exclude the name so this reads: `Use `typing.TypeAlias` for type aliases. e.g. "AliasNone: typing.TypeAlias = None"`

```suggestion
        format!("Use `typing.TypeAlias` for type aliases, e.g. \"{name}: typing.TypeAlias = {value}\"")
```

or make it singular:  `Use `typing.TypeAlias` for type alias `AliasNone`, e.g. "AliasNone: typing.TypeAlias = None"